src/main/scala/Engine.scala

```diff
@@ -41,8 +41,9 @@ case class Query(
     returnSelf: Option[Boolean] = None,// means for an item query should the item itself be returned, defaults
                                        // to what is in the algorithm params or false
     num: Option[Int] = None, // default: whatever is in algorithm params, which itself has a default--probably 20
-    eventNames: Option[List[String]]) // names used to ID all user actions
-  extends Serializable
+    eventNames: Option[List[String]], // names used to ID all user actions
+    addRank: Option[Boolean] = None // Add to ItemScore rank fields values
+) extends Serializable
 
```

Comment:
also this mehod allows the user to define the names of the field in the index as is done with popRank already